dep ensure fails on non-git repositories

go version go1.7.3 windows/amd64

Steps to reproduce:

1. `go get github.com/ChrisHines/dep-test`
2. `dep init`
3. `dep ensure`

Expected to see package bitbucket.org/zombiezen/gopdf/pdf in the vendor folder.

Instead I got this error:

```
C:\Users\Chris\Go\src\github.com\ChrisHines\dep-test>dep init

C:\Users\Chris\Go\src\github.com\ChrisHines\dep-test>dep ensure
grouped write of manifest, lock and vendor: error while writing out vendor tree: error while exporting bitbucket.org/zombiezen/gopdf: destination already exists
```

Additional hints:

I first encountered this on a bigger (private) project. The error can also occur for packages hosted in Bazaar repos on launchpad.net. Hacking on the dep code to dump out additional info I traced the origin of the error to https://github.com/golang/dep/blob/master/vendor/github.com/sdboyer/gps/util.go#L88 which is called a few stack frames down from https://github.com/golang/dep/blob/master/vendor/github.com/sdboyer/gps/result.go#L51. In short `WriteDepTree` makes the destination directory, but the implementation of interface method `source.exportVersionTo` for non-git repos calls `copyDir` which requires the destination not to exist.

I'm starting to follow up on this issue. I added some debug prints, rebuilt, and ... nothing changed.  Hmmm, I got bit by building with `go install -v github.com/golang/dep` rather than `go install -v github.com/golang/dep/cmd/dep`, so my dep binary wasn't getting rebuilt. It looks like that may have been the case on my most recent reports above as well.

So here is the current status, _actually_ building 2c974c5ecb5394a2c39f7c23dc2e594816383f9d:

```
C:\Users\Chris\Go\src\github.com\ChrisHines\dep-test>dep init -v
dep: Finding dependencies for "github.com/ChrisHines/dep-test"...
dep: Found 1 dependencies.
dep: Building dependency graph...
dep: Found import of "bitbucket.org/zombiezen/gopdf/pdf", analyzing...
dep: Could not determine version for "bitbucket.org/zombiezen/gopdf", omitting from generated manifest
dep: Found import of "launchpad.net/gocheck", analyzing...
dep: Could not determine version for "launchpad.net/gocheck", omitting from generated manifest
dep: Analyzing transitive imports...
dep: Analyzing "bitbucket.org/zombiezen/gopdf/pdf"...
dep: Analyzing "launchpad.net/gocheck"...
dep: Solving...
Cached bitbucket.org/zombiezen/gopdf
Cached launchpad.net/gocheck
Root project is "github.com/ChrisHines/dep-test"
 1 transitively valid internal packages
 2 external packages imported from 2 projects
(0)   ✓ select (root)
(1)     ? attempt bitbucket.org/zombiezen/gopdf with 1 pkgs; 0 versions to try
panic: version queue is empty, should not happen

goroutine 1 [running]:
github.com/golang/dep/gps.(*solver).findValidVersion(0xc0421a65a0, 0xc042047400, 0xc04216b100, 0x1, 0x1, 0x0, 0xc04216b100)
        C:/Users/Chris/Go/src/github.com/golang/dep/gps/solver.go:804 +0x601
github.com/golang/dep/gps.(*solver).createVersionQueue(0xc0421a65a0, 0xc04214a6a0, 0x1d, 0x0, 0x0, 0xc04216b100, 0x1, 0x1, 0x0, 0x0, ...)
        C:/Users/Chris/Go/src/github.com/golang/dep/gps/solver.go:791 +0xb03
github.com/golang/dep/gps.(*solver).solve(0xc0421a65a0, 0x0, 0x0, 0x1e)
        C:/Users/Chris/Go/src/github.com/golang/dep/gps/solver.go:408 +0x478
github.com/golang/dep/gps.(*solver).Solve(0xc0421a65a0, 0x34, 0xa6bd00, 0xab9608, 0xc0420fbf60)
        C:/Users/Chris/Go/src/github.com/golang/dep/gps/solver.go:359 +0xb2
main.(*initCommand).Run(0xc042129421, 0xc042112c00, 0xc042054290, 0x0, 0x0, 0x0, 0x0)
        C:/Users/Chris/Go/src/github.com/golang/dep/cmd/dep/init.go:166 +0xd6d
main.main()
        C:/Users/Chris/Go/src/github.com/golang/dep/cmd/dep/main.go:128 +0x6f9
```

--- a/gps/source.go
+++ b/gps/source.go
@@ -464,7 +464,7 @@ func (sg *sourceGateway) require(ctx context.Context, wanted sourceState) (errSt
 					return err
 				})
 
-				if err != nil {
+				if err == nil {
 					sg.cache.storeVersionMap(pvl, true)
 				}
 			case sourceHasLatestLocally:

